frontend/src/tests/lib/routes/NeuronDetail.spec.ts

```
@@ -0,0 +1,132 @@
+} from "../../mocks/sns-neurons.mock";
+import { snsResponseFor } from "../../mocks/sns-response.mock";
+import { NeuronDetailPo } from "../../page-objects/NeuronDetail.page-object";
+import { blockAllCallsTo } from "../../utils/module.test-utils";
+
+jest.mock("$lib/api/governance.api");
+jest.mock("$lib/api/sns-governance.api");
+jest.mock("$lib/api/sns-ledger.api");
+jest.mock("$lib/api/sns.api");
+
+const blockedApiPaths = [
+  "$lib/api/governance.api",
+  "$lib/api/sns.api",
+  "$lib/api/sns-ledger.api",
+  "$lib/api/sns-governance.api",
+];
+
+const testSnsCanisterId = Principal.fromHex("123321");
+const testFee = BigInt(25000);
+const testSnsNeuronId = getSnsNeuronIdAsHexString(mockSnsNeuron);
+const testNnsNeuronId = mockNeuron.neuronId;
+
+const snsProps = {
+  neuronId: testSnsNeuronId,
+};
+
+const nnsProps = {
+  neuronId: testNnsNeuronId,
+};
+
+describe("NeuronDetail", () => {
+  blockAllCallsTo(blockedApiPaths);
+
+  beforeEach(() => {
+    snsQueryStore.reset();
+  });
+
+  describe("nns neuron", () => {
+    beforeEach(() => {
+      jest.mocked(governanceApi.queryNeurons).mockResolvedValue([mockNeuron]);
+      jest.mocked(governanceApi.queryKnownNeurons).mockResolvedValue([]);
+      page.mock({
+        data: { universe: OWN_CANISTER_ID_TEXT },
+        routeId: AppPath.Neuron,
+      });
+    });
+
+    it("should load", async () => {
+      const { container } = render(NeuronDetail, nnsProps);
+      const po = NeuronDetailPo.under(container);
+
+      expect(po.hasSnsNeuronDetail()).toBe(false);
+      expect(po.hasNnsNeuronDetail()).toBe(true);
+      expect(po.getNnsNeuronDetail().isContentLoaded()).toBe(true);
+    });
+  });
+
+  describe("sns neuron", () => {
+    beforeEach(() => {
+      const [metadata, swapStates] = snsResponseFor({
+        principal: testSnsCanisterId,
+        lifecycle: SnsSwapLifecycle.Committed,
+      });
+      jest.mocked(snsApi.queryAllSnsMetadata).mockResolvedValue(metadata);
```

Careful, when we use the SNS aggregator this is not called.

I guess that because the aggregator fails, the code fallsback here. But ideally we'll remove the fallback.

It makes more sense to mock the call to the `sns-aggregator.api`

---

I thought all featureFlags are enabled by default in tests. But in that case it should have already use the aggregator, no?

---

It was actually enabled but it was using the fallback after the aggregator call failed silently. Now I'm mocking the aggregator.

---

great, thanks!